algosdk/v2client/algod.py

```diff
@@ -397,16 +397,58 @@ def genesis(self, **kwargs):
         req = "/genesis"
         return self.algod_request("GET", req, **kwargs)
 
-    def proof(self, round_num, txid, **kwargs):
+    def transaction_proof(
+        self, round_num, txid, hashtype="", response_format="json", **kwargs
+    ):
         """
         Get the proof for a given transaction in a round.
 
         Args:
             round_num (int): The round in which the transaction appears.
             txid (str): The transaction ID for which to generate a proof.
+            hashtype (str): The state proof hash type, sha256 or sha512_256.
+            response_format (str): the format in which the response is returned: either
+                "json" or "msgpack"
         """
+        params = {"format": response_format}
+        if hashtype != "":
+            params["hashtype"] = hashtype
         req = "/blocks/{}/transactions/{}/proof".format(round_num, txid)
-        return self.algod_request("GET", req, **kwargs)
+        return self.algod_request(
+            "GET",
+            req,
+            params=params,
+            response_format=response_format,
+            **kwargs
+        )
+
+    def lightheader(self, round_num, response_format="json", **kwargs):
+        """
+        Gets a proof for a given light block header inside a state proof commitment.
+
+        Args:
+            round_num (int): The round in which the transaction appears.
+            response_format (str): the format in which the response is returned: either
+             "json" or "msgpack"
+        """
+        req = "/blocks/{}/lightheader/proof".format(round_num)
+        return self.algod_request(
+            "GET", req, response_format=response_format, **kwargs
+        )
+
+    def stateproofs(self, round_num, response_format="json", **kwargs):
+        """
+        Get a state proof that covers a given round.
+
+        Args:
+            round_num (int): The round in which the transaction appears.
```

Comment:
This is not completely accurate. The API would search for a state proof that covers the given round. The state proof txn would not appear there.